test/orchestratord/mzcompose.py

```diff
@@ -1930,23 +1946,50 @@ def run(definition: dict[str, Any], expect_fail: bool) -> None:
 
 
 def post_run_check(definition: dict[str, Any], expect_fail: bool) -> None:
-    for i in range(60):
+    for i in range(900):
         try:
-            spawn.capture(
-                [
-                    "kubectl",
-                    "get",
-                    "materializes",
-                    "-n",
-                    "materialize-environment",
-                ],
-                stderr=subprocess.DEVNULL,
+            data = json.loads(
+                spawn.capture(
+                    [
+                        "kubectl",
+                        "get",
+                        "materializes",
+                        "-n",
+                        "materialize-environment",
+                        "-o",
+                        "json",
+                    ],
+                    stderr=subprocess.DEVNULL,
+                )
             )
-            break
+            status = data["items"][0].get("status")
+            if not status:
+                continue
+            if (
+                not status["conditions"]
+                or status["conditions"][0]["type"] != "UpToDate"
+            ):
+                continue
+            if (
+                status["lastCompletedRolloutRequest"]
+                == data["items"][0]["spec"]["requestRollout"]
+            ):
+                break
         except subprocess.CalledProcessError:
             pass
         time.sleep(1)
```

Comment:
This sleep doesn't happen if we short circuit in the `continue` statements above. This causes it to loop very fast, and not complete sometimes.